### PR TITLE
feat: add extensions page in settings

### LINKED ISSUE #3991: Extensions page in settings page that lists currently installed plugins
Go to `/settings/extensions` to see list of extensions that are in the extension-manifest database.

If an extension is in the database, but not installed, show some special UI.

Currently, we will just show the pre-installed extensions. We will eventually hide those from the user.

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
## Description

Adds extensions page that shows installed extensions. Users can search for extensions and install, enable/disable and uninstall.

## Task Item

closes #3991

## Screenshots

image
image

### CODE REVIEW

File: Composer/packages/extension/src/types/extension.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+export type ExtensionContributionPage = {
+  /** Id of cooresponding UI bundle. */
+  bundleId: string;
+  /** Label to dispaly in nav. */
+  label: string;
+  /** Optional icon to use in nav. Available icons [here](https://developer.microsoft.com/en-us/fluentui#/styles/web/icons). */
```
Comment (tonyanziano) [inline Composer/packages/extension/src/types/extension.ts:9]: 👍

File: Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  CheckboxVisibility,
+} from 'office-ui-fabric-react/lib/DetailsList';
+import { ShimmeredDetailsList } from 'office-ui-fabric-react/lib/ShimmeredDetailsList';
+import axios from 'axios';
+import formatMessage from 'format-message';
+
+import httpClient from '../../../utils/httpUtil';
+
+// TODO: extract to shared?
+type ExtensionSearchResult = {
```
Comment (tonyanziano) [inline Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx:23]: this is being used in `@bfc/extension` as well, so might as well move it to shared

**Review by tonyanziano — CHANGES_REQUESTED** [review]
A few minor comments. Looks great otherwise 👍

File: Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx (hunk 2 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx:23]: My plan is to create an extensions shared package eventually. So I'm fine with some duplication in the meantime.

File: Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      dialogContentProps={{
+        type: DialogType.close,
+        title: formatMessage('Add new extension'),
+      }}
+      hidden={!isOpen}
+      minWidth="800px"
+      onDismiss={onDismiss}
+    >
+      <div>
+        <TextField
```
[the remark this answers is not in the document]
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx:137]: Good suggestion. I already have this change but haven't pushed yet.

File: Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+const InstallExtensionDialog: React.FC<InstallExtensionDialogProps> = (props) => {
+  const { isOpen, onDismiss, onInstall } = props;
+  const [searchQuery, setSearchQuery] = useState<string | null>(null);
+  const [matchingExtensions, setMatchingExtensions] = useState<ExtensionSearchResult[]>([]);
+  const [selectedExtension, setSelectedExtension] = useState<ExtensionSearchResult | null>(null);
+  const [isSearching, setIsSearching] = useState(false);
+
+  useEffect(() => {
+    if (searchQuery !== null) {
```
[the remark this answers is not in the document]
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx:45]: I will perform a search without a query. Null here represents no user input yet.

File: Composer/packages/extension/src/manager/manager.ts [hunk 5]
```diff
@@ -286,6 +281,43 @@ class ExtensionManager {
...
+            }
+          });
+        }
+      } catch (err) {
+        log('%O', err);
+      }
+    }
+  }
+
+  private async ensureRemoteDir() {
```
[the remark this answers is not in the document]
Comment (a-b-r-o-w-n) [reply Composer/packages/extension/src/manager/manager.ts:316]: Oh nice. I'll update.

File: Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (hatpick) [reply Composer/packages/client/src/pages/setting/extensions/InstallExtensionDialog.tsx:54]: Sounds good, it was just a nit :D

**Review by tonyanziano — COMMENTED** [review]
2 strings that need to be localized -- other than that looks good 👍

**Review by tonyanziano — APPROVED** [review]

**Pull request merged**